mailet/api/src/main/java/org/apache/mailet/Mail.java

```diff
@@ -98,7 +99,9 @@
      * @throws MessagingException when an error occurs while retrieving the message
      */
     MimeMessage getMessage() throws MessagingException;
-    
+
+    Message getParsedMimeMessage() throws MessagingException;
```

Comment:
two cases: it can be not parsed, or it can need to be reparsed (case where original MimeMessage changed). Maybe a more friendly happy would be to handle the state inside, and parse it anytime getParsedMimeMessage is called (and not already modified). I know that many api do this this way, even if it hides the cost of parsing.